lightning/src/onion_message/messenger.rs

```diff
@@ -1047,18 +1081,36 @@ where
 	/// enqueueing any response for sending.
 	pub fn handle_onion_message_response<T: OnionMessageContents>(
 		&self, response: ResponseInstruction<T>
-	) {
-		if let ResponseInstruction::WithoutReplyPath(response) = response {
-			let message_type = response.message.msg_type();
-			let _ = self.find_path_and_enqueue_onion_message(
-				response.message, Destination::BlindedPath(response.reply_path), None,
-				format_args!(
-					"when responding with {} to an onion message with path_id {:02x?}",
-					message_type,
-					response.path_id
-				)
-			);
-		}
+	) -> Result<Option<SendSuccess>, SendError> {
+		let (response, create_reply_path) = match response {
+			ResponseInstruction::WithReplyPath(response) => (response, true),
+			ResponseInstruction::WithoutReplyPath(response) => (response, false),
+			ResponseInstruction::NoResponse => return Ok(None),
+		};
+
+		let message_type = response.message.msg_type();
+		let log_suffix = format!(
+			"when responding with {} to an onion message with path_id {:02x?}",
+			message_type,
+			response.path_id
+		);
```

Comment:
We may actually want to inline this in both places using `format_args!`. Otherwise, we are allocating a string even if the user's log level doesn't trigger any logging.